### DISCUSSION TITLE
Debian/Ubuntu repositories for LTS/Current release channels

### DISCUSSION THREAD

**Discussion by ferferga** [nodesource/distributions#1796] [Ideas]
Right now, all the available repos are for specific versions. For instance, running the script located at ``https://deb.nodesource.com/setup_lts.x`` will set up the following ``nodesource.list`` file:

```
deb [signed-by=/etc/apt/keyrings/nodesource.gpg] https://deb.nodesource.com/node_20.x nodistro main
```

It would be great if a ``https://deb.nodesource.com/node_lts nodistro main`` and a ``https://deb.nodesource.com/node_current nodistro main`` existed, [like Debian with it's ``stable`` channel](https://wiki.debian.org/DebianStable)

**Comment by blueray453** [comment]
I came here to say the same thing. Thank you very much.

**Comment by jehon** [comment]
Yes, that would be awesome, and simplify the repository maintenance on my servers too

**Comment by riosje** [comment]
Hi Guys how would you handle the breaking changes between every major version?
At some point you will get a new major version with breaking changes and your Node.js process might suddenly stop working due to some compatibility issues with the new version.
I think we can implement this but want to be sure this will not become in an issue in the future.

**Reply by ferferga** [reply]
@riosje That's not something you (as maintainers of this project) should worry about, in my opinion. The best example of this is already the ``stable`` apt repo that Debian provides, which allows you to upgrade between major versions. But before performing the upgrade, ``apt upgrade`` gives me the notice that a new version is available and allows me to take any action (which in this case would be to review it and perform necessary upgrades)

Docker images and [GitHub's ``actions/setup-node``](https://github.com/actions/setup-node?tab=readme-ov-file#supported-version-syntax) already provides tags for lts or current (latest) versions, and it's up to each user the responsibility to choose wisely which one they need, not the maintainers of those distribution methods.

Also, I believe most active projects, nearing the start of the next LTS cycle, are already preparing for compatibility with the next LTS or have tests in their CI/CD pipelines to test the actual LTS and the current version.

**Comment by jehon** [comment]
Yes, you are right that it is a risk, and so this would be mainly for dev, acceptance and CI... I would like to see breaking changes as soon as they appear rather than having my setup break later on.

And a good point would be to mention that risk in the documentation.

**Comment by Gasantan2** [comment]
Try some of the NVM repo for node.
